lib/spare_keys.rb

```diff
@@ -26,6 +26,15 @@ def self.use_keychain(keychain_path, clear_list = false, type = nil, domain = ni
                 yield if block_given?
             ensure
                 original_keychain = `security #{type}-keychain #{domain_flag} -s #{original_keychain}` if type
+                
+                unless clear_list
+                    # Grab the keychain list as it looks right now in case
+                    # another process has changed it
+                    current_list = `security list-keychains #{domain_flag} | xargs`
+                    # Remove the supplied keychain
+                    original_list = (current_list.split(" ").reject { | item | item.include? keychain_path }).join(" ")
```

Comment:
I have two concerns here:

1. Splitting on space could cause issues if there is a space in the path. I know that OSX tends to push you away from having a space in your username, but can it be done anyway? Or, alternatively, can the default keychain path be changed to somewhere that _does_ have a space? Perhaps transforming the raw (ie. minus xargs) output might be safer
2. `use_keychain` doesn't require a full path to the keychain, so `login` is a perfectly valid value. However, it would also match `my_awesome_login` or  `/alt/path/to/login.keychain`, which would cause unexpected behavior.